File: problems/Interval List Intersections/intervalIntersection.cs
public class Solution {
    public int[][] IntervalIntersection(int[][] A, int[][] B) {
        var result = new List<int[]>();
        var aIdx = 0;
        var bIdx = 0;

        while (A.Length > aIdx && B.Length > bIdx) {
            var x = Math.Max(A[aIdx][0], B[bIdx][0]);
            var y = Math.Min(A[aIdx][1], B[bIdx][1]);

            if (x <= y) {
                result.Add(new [] {x, y});
            }

            if (A[aIdx][1] < B[bIdx][1]) {
                ++aIdx;
            } else {
                ++bIdx;
            }
        }

        return result.ToArray();
    }
}

File: problems/Interval List Intersections/intervalIntersection.java
class Solution {
    public int[][] intervalIntersection(int[][] firstList, int[][] secondList) {
        List<int[]> result = new ArrayList<>();
        int aIdx = 0;
        int bIdx = 0;

        while (firstList.length > aIdx && secondList.length > bIdx) {
            int x = Math.max(firstList[aIdx][0], secondList[bIdx][0]);
            int y = Math.min(firstList[aIdx][1], secondList[bIdx][1]);

            if (x <= y) {
                result.add(new int[] {x, y});
            }

            if (firstList[aIdx][1] < secondList[bIdx][1]) {
                ++aIdx;
            } else {
                ++bIdx;
            }
        }

        return result.toArray(new int[result.size()][2]);
    }
}
